### PR TITLE
refine cmd/agent/main.go and cmd/server/main.go

### PR DESCRIPTION
**Pull request by liufen90**
refine cmd/agent/main.go and cmd/server/main.go

### CODE REVIEW

File: cmd/agent/app/options/options.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		ProbeInterval:           o.ProbeInterval,
+		SyncIntervalCap:         o.SyncIntervalCap,
+		DialOptions:             dialOptions,
+		ServiceAccountTokenPath: o.ServiceAccountTokenPath,
+	}
+}
+
+func (o *GrpcProxyAgentOptions) Flags() *pflag.FlagSet {
+	flags := pflag.NewFlagSet("proxy-agent", pflag.ContinueOnError)
+	flags.StringVar(&o.AgentCert, "agent-cert", o.AgentCert, "If non-empty secure communication with this cert.")
```
Comment (cheftako) [inline cmd/agent/app/options/options.go:63]: I realize this is largely a forklift but slightly nervous about making sure all the flags are maintained (i.e. binaries are backward compatible). Any chance of tests for the Flags functions?

Comment (liufen90) [reply cmd/agent/app/options/options.go:63]: I checked that I didn't modify the flags' names. And I tested the flags locally.

**Comment by liufen90** [comment]
@cheftako  please have a look

File: cmd/server/app/server.go [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+			fmt.Fprintf(w, "ok")
+			return
+		}
+		w.WriteHeader(500)
+		fmt.Fprintf(w, msg)
+	})
+
+	muxHandler := http.NewServeMux()
+	muxHandler.HandleFunc("/healthz", livenessHandler)
+	muxHandler.HandleFunc("/ready", readinessHandler)
```
Comment (cheftako) [inline cmd/server/app/server.go:355]: I realize this was a pre-existing bug but this should be ```/readyz```.

File: cmd/agent/app/server.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+		fmt.Fprintf(w, "ok")
+	})
+	readinessHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "ok")
+	})
+
+	muxHandler := http.NewServeMux()
+	muxHandler.Handle("/metrics", promhttp.Handler())
+	muxHandler.HandleFunc("/healthz", livenessHandler)
+	muxHandler.HandleFunc("/ready", readinessHandler)
```
Comment (cheftako) [inline cmd/agent/app/server.go:85]: I realize this was a pre-existing bug but this should be ```/readyz```.

File: cmd/server/app/server.go (hunk 2 above)
Comment (liufen90) [reply cmd/server/app/server.go:355]: I found that it is /ready in master branch. Maybe we can fix it in other PRs. This PR only refine main package.

File: cmd/agent/app/server.go (hunk 3 above)
Comment (liufen90) [reply cmd/agent/app/server.go:85]: I found that it is /ready in master branch. Maybe we can fix it in other PRs. This PR only refine main package.

**Pull request merged**